File: ToyMachineInterpreterFX/src/Model/FileStatement/IFileTable.java
package Model.FileStatement;

import java.util.Collection;

public interface IFileTable<K, V> {
    void add(K key, V value);
    void delete(K key);
    V get(K key);
    boolean contains(K key);
    Iterable<K> getAllKeys();
    Collection<V> getAllValues();
}

File: Toy-Language-C#-Intepreter/Toy-Language-CS-Interpreter/Toy-Language-CS-Interpreter/Model/Containers/IFileTable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Toy_Language_CS_Interpreter.Model.Containers
{
    interface IFileTable<K,V>
    {
        void Add(K key, V value);
        void Delete(K key);
        V Get(K key);
        bool Contains(K key);
        List<K> GetAllKeys();
        List<V> GetAllValues();
    }
}
